lib/BuildSystem/BuildSystem.cpp

```diff
@@ -2074,7 +2075,24 @@ class ShellCommand : public ExternalCommand {
       virtual void actOnRuleDependency(const char* dependency,
                                        uint64_t length,
                                        const StringRef unescapedWord) override {
-        bsci.taskDiscoveredDependency(task, BuildKey::makeNode(unescapedWord));
+        if (llvm::sys::path::is_absolute(unescapedWord)) {
+          bsci.taskDiscoveredDependency(task, BuildKey::makeNode(unescapedWord));
+          return;
+        }
+
+        // Generate absolute path
+        //
+        // NOTE: This is making the assumption that relative paths coming in a
+        // dependency file are in relation to the explictly set working
+        // directory, or the current working directory when it has not been set.
+        SmallString<PATH_MAX> absPath = StringRef(command->workingDirectory);
```

Comment:
Hmm... this isn't quite absolute just yet.  It is assuming that 'workingDirectory' is absolute, which may not be the case.

Reply:
Are we sure this isn't a clang bug? If clang generates relative paths, and something uses -fworking-directory, could we ever handle that case correctly?

Reply:
In any case, I agree we should handle this case, just wondering if there is a corner case we can't fix

Reply:
Should we just make sure that every command's working directory is absolute?

Reply:
Probably not a bad idea.

Reply:
Agreed, I don't think it makes any sense to allow relative working directories given that the manifest is generally intended to be entirely non-relocatable.